clang/lib/Sema/SemaDecl.cpp

```diff
@@ -6888,6 +6888,42 @@ static void SetNestedNameSpecifier(Sema &S, DeclaratorDecl *DD, Declarator &D) {
   DD->setQualifierInfo(SS.getWithLocInContext(S.Context));
 }
 
+void Sema::deduceHLSLAddressSpace(VarDecl *Decl) {
```

Comment:
Could you please move this code to `SemaHLSL::deduceAddressSpace(VarDecl *Decl)` and call it from directly `SemaHLSL::ActOnVariableDeclarator`?